js/src/langchain.ts

```diff
@@ -113,7 +119,40 @@ export class RunnableTraceable<RunInput, RunOutput> extends Runnable<
 
   async invoke(input: RunInput, options?: Partial<RunnableConfig>) {
     const [config] = this._getOptionsList(options ?? {}, 1);
-    return (await this.func(config, input)) as RunOutput;
+    const callbacks = await getCallbackManagerForConfig(config);
+
+    return (await this.func(
+      patchConfig(config, { callbacks }),
+      input
+    )) as RunOutput;
+  }
+
+  async *_streamIterator(
+    input: RunInput,
+    options?: Partial<RunnableConfig>
+  ): AsyncGenerator<RunOutput> {
+    const result = await this.invoke(input, options);
+
+    if (isAsyncIterable(result)) {
+      const iterator = result[Symbol.asyncIterator]();
+      while (true) {
+        const { done, value } = await iterator.next();
+        if (done) break;
+        yield value as RunOutput;
+      }
+      return;
+    }
+
+    if (isIteratorLike(result)) {
```

Comment:
What's the use case for this check? Seems unnecessary?

Reply:
If the traceable returns a generator, the stream wouldn't caught that